### DISCUSSION TITLE
Use Stitches native responsive variants

### DISCUSSION THREAD

**Discussion by hmbrg** [heroui-inc/heroui#168] [Help] [answered]
Right now, NextUI doesn't use Stitches built-in [responsive variants feature](https://stitches.dev/docs/responsive-styles#responsive-variants). 
This makes using responsive props for [components a bit awkward](https://github.com/nextui-org/nextui/discussions/164) to use and not support SSR.

Stitches built-in responsive variants work using CSS media queries and don't use any JS under the hood. Using them looks like this: 
```javascript
() => (
  <Button
    color={{
      '@initial': 'gray',
      '@bp2': 'violet'
    }}
  >
    Button
  </Button>
);
```

Thanks for the amazing library! This is the missing layer above Stitches that I've dreamed about for a long time. I will definitely use it for my next project.

**Comment by jrgarciadev** [comment, marked as answer]
Hey @hmbrg I'm glad you liked it!, actually NextUI does have built-in Stitches media queries, I just cleared that answer -> https://github.com/nextui-org/nextui/discussions/164#discussioncomment-2063208, it was just a misunderstanding 

Hope this helps 👍🏻

**Reply by hmbrg** [reply]
Hey @jrgarciadev, thanks for your answer! I figured out that I can use Stitches media queries inside the `css` prop but what is not working for me, since the component types won't allow it, is using Stitches responsive variants feature. As showed in my example, that would allow for any variant props (like `color` — not just `css`) to be applied based on media queries. Hope this clears it up! :)
